parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -503,6 +512,44 @@ ensureKubeCACert() {
     chmod 0600 "${KUBE_CA_FILE}"
 }
 
+# drop-in path defined outside so configureAndStartSecureTLSBootstrapping can be unit tested
+SECURE_TLS_BOOTSTRAPPING_DROP_IN="/etc/systemd/system/secure-tls-bootstrap.service.d/10-securetlsbootstrap.conf"
+configureAndStartSecureTLSBootstrapping() {
+    mkdir -p "$(dirname "${SECURE_TLS_BOOTSTRAPPING_DROP_IN}")"
+    touch "${SECURE_TLS_BOOTSTRAPPING_DROP_IN}"
+    chmod 0600 "${SECURE_TLS_BOOTSTRAPPING_DROP_IN}"
+    cat > "${SECURE_TLS_BOOTSTRAPPING_DROP_IN}" <<EOF
+[Service]
+Environment="BOOTSTRAP_FLAGS=--aad-resource="${CUSTOM_SECURE_TLS_BOOTSTRAP_AAD_SERVER_APP_ID:-6dae42f8-4368-4678-94ff-3960e28e3630}" --apiserver-fqdn=${API_SERVER_NAME} --cloud-provider-config=${AZURE_JSON_PATH}"
+EOF
+
+    systemctlEnableAndStartNoBlock secure-tls-bootstrap 30 || exit $ERR_SECURE_TLS_BOOTSTRAP_START_FAILURE
+}
+
+# kubeconfig path defined outside so ensureSecureTLSBootstrapping can be unit tested
+KUBECONFIG_PATH="/var/lib/kubelet/kubeconfig"
+ensureSecureTLSBootstrapping() { 
+    while [ "$(systemctl is-active secure-tls-bootstrap)" = "activating" ]; do
+        echo "secure TLS bootstrapping is in-progress, waiting for terminal state..."
+        sleep 0.5   
+    done
+
+    if ! systemctl is-active secure-tls-bootstrap; then
```

Comment:
if not active we exits ?

Reply:
`! is-active` will catch the failed case and more (such as inactive if the unit wasn't ever able to be activated in the first place) - the RemainAfterExit=yes in the unit configuration will make it such that that bootstrap unit remains in an active state after it's completed, meaning it should be `active` by this time assuming everything succeeded